stacks/spica/src/bucket/components/relation-schema/relation-schema.component.ts

```diff
@@ -1,19 +1,36 @@
-import {Component, Inject} from "@angular/core";
+import {Component, Inject, AfterViewInit} from "@angular/core";
 import {INPUT_SCHEMA} from "@spica-client/common";
 import {Observable} from "rxjs";
 import {BucketService} from "../../services/bucket.service";
 import {Bucket} from "../../interfaces/bucket";
-import {RelationSchema} from "../relation";
+import {RelationSchema, RelationType} from "../relation";
+import {tap} from "rxjs/operators";
 
 @Component({
   selector: "bucket-relation",
   templateUrl: "./relation-schema.component.html",
   styleUrls: ["./relation-schema.component.scss"]
 })
-export class RelationSchemaComponent {
+export class RelationSchemaComponent implements AfterViewInit {
   public buckets: Observable<Bucket[]>;
 
-  constructor(@Inject(INPUT_SCHEMA) public schema: RelationSchema, bucketService: BucketService) {
-    this.buckets = bucketService.getBuckets();
+  constructor(
+    @Inject(INPUT_SCHEMA) public schema: RelationSchema,
+    private bucketService: BucketService
+  ) {}
+
+  ngAfterViewInit() {
+    setTimeout(() => {
+      if (!this.schema.relationType) {
+        this.schema.relationType = RelationType.OneToOne;
+      }
+      this.buckets = this.bucketService.getBuckets().pipe(
+        tap(buckets => {
+          if (!this.schema.bucketId) {
+            this.schema.bucketId = buckets[0]._id;
+          }
+        })
+      );
```

Comment:
You can put these into constructor or ngOnInit. then you wouldn't need to use setTimeout.

Reply:
**Expression has changed after it was checked** error shows up when I made these changes.

Reply:
How about doing it in the constructor?

Reply:
Same error.

Reply:
It works in the constructor with setTimeout.